### PR TITLE
feat: Refactoring

### PR DESCRIPTION
**Pull request by JakWai01**
Closes https://github.com/alphahorizonio/consumat.io/issues/122
Closes https://github.com/alphahorizonio/consumat.io/issues/128
Closes https://github.com/alphahorizonio/consumat.io/issues/114

### CODE REVIEW

File: consumatio/usecases/list.py [hunk 1]
```diff
@@ -4,21 +4,32 @@
...
+    if watchStatus == "any" and favorite:
+        query += " AND media_data.favorite_content = :favorite AND media_data.media_type_content = :type AND user_data.external_id_content = :user;"
+        results = MediaData.query.from_statement(text(query)).params(
+            favorite=favorite, type=type, user=external_id).all()
+    elif watchStatus == "any":
+        query += " AND media_data.media_type_content = :type AND user_data.external_id_content = :user;"
+        results = MediaData.query.from_statement(text(query)).params(
+            type=type, user=external_id).all()
+    elif favorite:
+        query += " AND media_data.watch_status_content = :watch_status AND media_data.favorite_content = :favorite AND media_data.media_type_content = :type AND user_data.external_id_content = :user;"
```
Comment (pojntfx) [inline consumatio/usecases/list.py:23]: Just double checking: This isn't a SQL injection as the only parameters passed are in the `from_statement` call , right? Does this sanitize the input strings?

**Review by pojntfx — CHANGES_REQUESTED** [review]
Just one quick clarification concerning a potential SQL injection :)

File: consumatio/usecases/list.py (hunk 1 above)
Comment (JakWai01) [reply consumatio/usecases/list.py:23]: I think when using params the strings get sanitized although I am not 100% sure. We should probably convert the textual queries to "ORM Queries" ASAP. I'd like to handle this in a different PR though.

Comment (pojntfx) [reply consumatio/usecases/list.py:23]: Sounds good :+1:

**Pull request merged**